packages/runtime-common/realm.ts

```diff
@@ -658,6 +673,47 @@ export class Realm {
     }
     return data;
   }
+
+  private listeningClients: WritableStream[] = [];
+
+  private async subscribe(req: Request): Promise<Response> {
+    let headers = {
+      "Content-Type": "text/event-stream",
+      "Cache-Control": "no-cache",
+      Connection: "keep-alive",
+    };
+
+    let { response, writable } = this.#adapter.createStreamingResponse(
+      req,
+      {
+        status: 200,
+        headers,
+      },
+      () => {
+        this.listeningClients = this.listeningClients.filter(
+          (w) => w !== writable
+        );
+        this.sendUpdateMessages("clean up called");
+      }
+    );
+
+    this.listeningClients.push(writable);
+    this.sendUpdateMessages("updated clients");
+    // TODO: We may need to store something else here to do cleanup to keep
+    // tests consistent
+    waitForClose(writable);
+
+    return response;
+  }
+
+  private async sendUpdateMessages(message: string): Promise<void> {
```

Comment:
probably `message` is eventually going to be some kind of special type here where we have a type for all our messages like we do for the service worker messages